pkg/proxy/proxy.go

```diff
@@ -155,32 +150,138 @@ func (p *Proxy) SetIPContext(conn *networkservice.Connection, interfaceType netw
 			Name:   fmt.Sprintf("%s-dst", conn.Id),
 			Subnet: subnet,
 		}
-		dstPrefix, err := p.ipamClient.Allocate(context.TODO(), child)
+		dstPrefix, err := p.IpamClient.Allocate(ctx, child)
 		if err != nil {
 			return err
 		}
 		dstIpAddrs = append(dstIpAddrs, dstPrefix.ToString())
 	}
 
-	conn.GetContext().IpContext = &networkservice.IPContext{}
 	if interfaceType == networking.NSE {
-		conn.GetContext().IpContext.SrcIpAddrs = srcIPAddrs
-		conn.GetContext().IpContext.DstIpAddrs = dstIpAddrs
-		conn.GetContext().GetIpContext().ExtraPrefixes = p.bridge.GetLocalPrefixes()
+		p.setNSEIpContext(conn.GetContext().GetIpContext(), srcIPAddrs, dstIpAddrs)
 	} else if interfaceType == networking.NSC {
-		conn.GetContext().IpContext.SrcIpAddrs = dstIpAddrs
-		conn.GetContext().IpContext.DstIpAddrs = srcIPAddrs
+		conn.GetContext().GetIpContext().SrcIpAddrs = dstIpAddrs
+		conn.GetContext().GetIpContext().DstIpAddrs = srcIPAddrs
 	}
 	return nil
 }
 
-func (p *Proxy) UnsetIPContext(conn *networkservice.Connection, interfaceType networking.InterfaceType) error {
+func (p *Proxy) setNSEIpContext(ipContext *networkservice.IPContext, srcIPAddrs []string, dstIpAddrs []string) {
+	if len(ipContext.SrcIpAddrs) == 0 && len(ipContext.DstIpAddrs) == 0 { // First request
+		ipContext.SrcIpAddrs = srcIPAddrs
+		ipContext.DstIpAddrs = dstIpAddrs
+		ipContext.ExtraPrefixes = p.Bridge.GetLocalPrefixes()
+		return
+	}
+	// The request is an update
+	if contains(ipContext.ExtraPrefixes, p.Bridge.GetLocalPrefixes()) &&
+		contains(ipContext.SrcIpAddrs, srcIPAddrs) &&
+		contains(ipContext.DstIpAddrs, dstIpAddrs) {
+		return
```

Comment:
only matches if there are no VIPs, right?

Reply:
no, it tells if all seconds list is in first list, so VIP will be ignored.